### PR TITLE
feature sms and voice integrations 158174594

### PR DESCRIPTION
**Pull request by magicmarie**
What does this PR do?
Sets up healthchecks to be able to send out alerts concerning the cron jobs through SMS and voice. Right now, Healthchecks only sends alerts through email.

Description of Task to be completed?
The system is able to send out alerts through SMS and voice channels.

The files changed/created include:
Created:
add_twiliosms.html - Added UI for the page where the phone number is added
add_twiliovoice.html - Added UI for the page where the phone number is added

Edited
hc/api/transports.py - Added the SMS and voice models
hc/front/urls.py - Added the URL that redirects to the sms and voice configuration pages
hc/front/views.py - Added the function that loads the page where the number to sms is added
hc/templates/front/channels.html - Added sms and voice to the list of configurable channel options
hc/front/tests/test_add_channel.py - Added tests for sms and voice access

What are the relevant pivotal tracker stories?
[#158174594](https://www.pivotaltracker.com/n/projects/2177206)

### CODE REVIEW

**Comment by EdwinKato** [comment]
@magicmarie @mutebironald resolve conflicts

File: hc/api/transports.py [hunk 1]
```diff
@@ -4,10 +4,16 @@
...
 import requests
 from six.moves.urllib.parse import quote
+from twilio.rest import Client
 
 from hc.lib import emails
 
 
+#Twilio Account SID and AUTH_TOKEN details
+account_sid = "AC06a9064c4ccabad17ac56ade5054916c"
+auth_token = "f323d7a1411ef651daa54dacb4147dfb"
```
Comment (EdwinKato) [inline hc/api/transports.py]: Store these credentials in an environment. It's not good to commit these with your source code

File: hc/api/transports.py [hunk 2]
```diff
@@ -60,6 +66,23 @@ def notify(self, check):
         emails.alert(self.channel.value, ctx)
 
 
+class TwilioSms(Transport):
+    def notify(self, check):
+        message = client.messages.create(
```
Comment (EdwinKato) [inline hc/api/transports.py]: This variable `message` is not used anywhere

File: hc/api/transports.py [hunk 3]
```diff
@@ -4,10 +4,16 @@
...
 from six.moves.urllib.parse import quote
+from twilio.rest import Client
 
 from hc.lib import emails
 
 
+#Twilio Account SID and AUTH_TOKEN details
+account_sid = "AC06a9064c4ccabad17ac56ade5054916c"
+auth_token = "f323d7a1411ef651daa54dacb4147dfb"
+client = Client(account_sid, auth_token)
```
Comment (EdwinKato) [inline hc/api/transports.py]: `client` has 2 usages. Use OOP here. Consider putting it in the transport class

File: hc/front/tests/test_add_channel.py [hunk 4]
```diff
@@ -75,3 +77,27 @@ def test_bad_kinds_dont_work(self):
         self.client.login(username="alice@example.org", password="password")
         response = self.client.post(url, form)
         assert response.status_code == 400
+        
+    def test_twiliosms_works(self):
+        """ test sms intergration works"""
```
Comment (EdwinKato) [inline hc/front/tests/test_add_channel.py]: Rectify spelling of intergration

**Review by EdwinKato — CHANGES_REQUESTED** [review]

File: hc/api/transports.py [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
 import requests
 from six.moves.urllib.parse import quote
+from twilio.rest import Client
 
 from hc.lib import emails
 
 
+#Twilio Account SID and AUTH_TOKEN details
+# TWILIO_ACCOUNT_SID = os.environ.get("TWILIO_ACCOUNT_SID")
+# TWILIO_AUTH_TOKEN = os.environ.get("TWILIO_AUTH_TOKEN")
```
Comment (EdwinKato) [inline hc/api/transports.py]: Remove these commented out lines

File: hc/front/tests/test_add_channel.py [hunk 6]
```diff
@@ -75,3 +77,27 @@ def test_bad_kinds_dont_work(self):
         self.client.login(username="alice@example.org", password="password")
         response = self.client.post(url, form)
         assert response.status_code == 400
+        
+    def test_twiliosms_works(self):
+        """ test sms integration works"""
+        alice_channel = User.objects.get(email="alice@example.org")
+        alice_before = Channel.objects.filter(user=alice_channel).count()
+        self.client.login(username="bob@example.org", password="password")
+        url = "/integrations/add/"
```
Comment (EdwinKato) [inline hc/front/tests/test_add_channel.py]: Use reverse for `urls` always

**Review by EdwinKato — CHANGES_REQUESTED** [review]

File: hc/front/tests/test_add_channel.py (hunk 6 above)
Comment (mutebironald) [reply hc/front/tests/test_add_channel.py]: worked on this.

**Review by EdwinKato — APPROVED** [review]

**Pull request merged**